python/python/lance/lance/__init__.pyi

```diff
@@ -213,6 +214,7 @@ class _Dataset:
     def index_statistics(self, index_name: str) -> str: ...
     def serialized_manifest(self) -> bytes: ...
     def load_indices(self) -> List[Index]: ...
+    def describe_indexes(self) -> List[IndexDescription]: ...
```

Comment:
Naming nit: is there a reason we're going with "indexes" vs "indices"?

We probably should choose a plural form and be consistent everywhere. My preference would be "indices"